clients/python/girder_client/cli.py

```diff
@@ -50,67 +52,141 @@ def __init__(self, username, password, dryrun, blacklist,
-        help='use existing items of same name at same location or create a new'
-        ' one')
-    parser.add_argument(
-        '--blacklist', default='', required=False,
-        help='comma separated list of filenames to ignore')
-    parser.add_argument(
-        '--dryrun', action='store_true',
-        help='will not write anything to Girder, only report on what would '
-        'happen')
-    parser.add_argument('--api-url', required=False, default=None,
-                        help='full URL to the RESTful API of a Girder server')
-    parser.add_argument('--username', required=False, default=None)
-    parser.add_argument('--password', required=False, default=None)
-    parser.add_argument('--api-key', required=False, default=None)
-    parser.add_argument('--scheme', required=False, default=None)
-    parser.add_argument('--host', required=False, default=None)
-    parser.add_argument('--port', required=False, default=None)
-    parser.add_argument('--api-root', required=False, default=None,
-                        help='relative path to the Girder REST API')
-    parser.add_argument('-c', default='upload',
-                        choices=['upload', 'download', 'localsync'],
-                        help='command to run')
-    parser.add_argument('parent_id', help='id of Girder parent target')
-    parser.add_argument('--parent-type', required=False, default='folder',
-                        help='type of Girder parent target, one of ' +
-                        '(collection, folder, user)')
-    parser.add_argument('local_folder', help='path to local target folder')
-    parser.add_argument('--leaf-folders-as-items', required=False,
-                        action='store_true',
-                        help='upload all files in leaf folders'
-                        'to a single Item named after the folder')
-    args = parser.parse_args()
+parser = argparse.ArgumentParser(
+    prog='girder-cli', description='Perform common Girder CLI operations.')
+parser.add_argument('--username', required=False, default=None)
+parser.add_argument('--password', required=False, default=None)
+parser.add_argument('--api-key', required=False, default=None)
+parser.add_argument('--api-url', required=False, default=None,
+                    help='full URL to the RESTful API of a Girder server')
+parser.add_argument('--scheme', required=False, default=None)
+parser.add_argument('--host', required=False, default=None)
+parser.add_argument('--port', required=False, default=None)
+parser.add_argument('--api-root', required=False, default=None,
+                    help='relative path to the Girder REST API')
+parser.add_argument('--blacklist', default='', required=False,
+                    help='comma separated list of filenames to ignore'),
+parser.add_argument('--dryrun', action='store_true',
+                    help='will not write anything to Girder, only report'
+                    ' on what would happen'),
```

Comment:
I'd rather `--blacklist` and `--dryrun` be applied on the subparsers since there's no guarantee that they'll be sensible options on all future subcommands.